### PR TITLE
feat: Migrate to poetry

### PR DESCRIPTION
**Pull request by Ankush-Pathak**
This will enable better dependency management and introduce poetry.lock.
List of dependencies with their specific versions and hashes in poetry.lock will aid vulnerability scanning.

This supersedes PR #4.

### CODE REVIEW

**Comment by j5awry** [comment]
changes like this require readme changes, as the mode of installation and development has changed. Please add documnetation

**Comment by j5awry** [comment]
Also, as stated in #4 , if we are going to change dependency management, our team has move heavily into `poetry`. What is the leading rationale to adopt `pip-tools` in this repo, and not `poetry`?

**Comment by Ankush-Pathak** [comment]
@j5awry I checked and a `poetry.lock` file with hashes would suffice the original request for these changes. I'll rework the PR.

**Comment by Ankush-Pathak** [comment]
@j5awry PR is ready for another round of review.

File: pyproject.toml [hunk 1]
```diff
@@ -1,55 +1,49 @@
...
-
-]
-license = {text = "GNU General Public License v3"}
-dependencies = [
-    "typer",
-    "launchpadlib",
-    "python-debian",
-    "rich",
-    "typing-extensions",
+  "Philip Roche <phil.roche@canonical.com>",
```
Comment (catmsred) [inline pyproject.toml:14]: You should add yourself at least to the maintainers list if not as an author

Comment (Ankush-Pathak) [reply pyproject.toml:14]: @catmsred Added myself to the maintainer list. :)

**Review by philroche — APPROVED** [review]
Nice

**Review by j5awry — APPROVED** [review]

**Pull request merged**